### PR TITLE
#167733646 login via Facebook and Google

### PR DESCRIPTION
**Pull request by chis0m**
#### What does this PR do?
create a login via facebook and google OAuth
#### Description of Task to be completed?
User should be able to create an account by login through
facebook and google account
#### How should this be manually tested?
After cloning the repo
start the app
Visit the endpoints for the social logins
#### What are the relevant pivotal tracker stories?
[#167733646](https://www.pivotaltracker.com/story/show/167733646)
#### Questions:

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -21,26 +21,11 @@
...
-    ]
+    "import/no-unresolved": [2, {
+      "commonjs": true
+    }],
+    "no-shadow": ["error", {
+      "allow": ["req", "res", "err"]
+    }]
   }
-}
+}
```
Comment (King-Success) [inline .eslintrc.json]: Add a line break here

File: src/config/config.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
-const dotenv = require('dotenv');
+// import dotenv from 'dotenv';
```
Comment (King-Success) [inline src/config/config.js]: Kindly remove all commented code

File: src/config/config.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
+require('dotenv').config();
 
-dotenv.config();
+// dotenv.config();
```
Comment (King-Success) [inline src/config/config.js]: remove commented code

File: src/migrations/20190821181942-create-user.js [hunk 4]
```diff
@@ -64,29 +64,41 @@ module.exports = {
...
+      unique: true,
+    },
+    profileImage: {
+      type: Sequelize.STRING
+    },
+    provider: {
+      type: Sequelize.STRING
     },
     department: {
       type: Sequelize.STRING,
```
Comment (King-Success) [inline src/migrations/20190821181942-create-user.js:104]: it is not good practice to edit a migration file after its been created. whenever you need to add, remove or adjust anything, you create a new migration file

File: .eslintrc.json (hunk 1 above)
Comment (emmanuelnwankwo) [inline .eslintrc.json]: Add line break at the bottom

File: .eslintrc.json [hunk 5]
```diff
@@ -21,26 +21,11 @@
...
-    "valid-jsdoc": [
-      "error",
-      {
-        "requireReturn": true,
-        "requireReturnType": true,
-        "requireParamDescription": false,
-        "requireReturnDescription": true
-      }
-    ],
-    "require-jsdoc": [
```
Comment (emmanuelnwankwo) [inline .eslintrc.json]: why did you remove the `require-jsdoc` and the above settings? Remember it was specified on eslint setup [task](https://www.pivotaltracker.com/story/show/167733632)

**Review by emmanuelnwankwo — CHANGES_REQUESTED** [review]
Great job @chis0m. I left some comments. Please share the values of the social login for .env in the team channel

File: .eslintrc.json (hunk 1 above)
Comment (chis0m) [reply .eslintrc.json]: > Add line break at the bottom

Ok thank you @emmanuelnwankwo

File: .eslintrc.json (hunk 5 above)
Comment (chis0m) [reply .eslintrc.json]: > why did you remove the `require-jsdoc` and the above settings? Remember it was specified on eslint setup [task](https://www.pivotaltracker.com/story/show/167733632)

Sorry @emmanuelnwankwo It must have been an oversight

**Review by King-Success — APPROVED** [review]

**Review by emmanuelnwankwo — APPROVED** [review]

File: src/services/UserService.js [hunk 6]
```diff
@@ -47,4 +47,30 @@ export default class UserService {
...
+   *
+   * @static
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @memberof SocialLogin
+   * @returns {object} - the user data object
+   *
+   */
+
+  static async socialLogin(userData) {
```
[the remark this answers is not in the document]
Comment (Baystef) [reply src/services/UserService.js:74]: I think you should have removed the non-async `find` method not the async, User.findOne is an asynchronous operation

File: src/migrations/20190821181942-create-user.js (hunk 4 above)
Comment (chis0m) [reply src/migrations/20190821181942-create-user.js:104]: ok noted

File: test/auth.spec.js [hunk 7]
```diff
@@ -229,4 +229,13 @@ describe('GET /api/auth/verify?token', () => {
       expect(response.body.data).to.be.a('string');
     });
   });
+
+  it('should sign in user if emaill is true', (done) => {
```
Comment (efeguono) [inline test/auth.spec.js:233]: I don't see how this tests the social authentication function

**Review by efeguono — CHANGES_REQUESTED** [review]

File: test/auth.spec.js (hunk 7 above)
Comment (chis0m) [reply test/auth.spec.js:233]: > I don't see how this tests the social authentication function

The idea is that the test endpoint has a middleware that mocks the facebook and google response data and passes it to the controller "socialLogin" which is the test candidate. Because to test the main URI, you have to make a call to the social API and receive the response data for a complete test cycle. As opposed to that you could just easily mock the response and pass it to the same controller and if it passes, it means the endpoint is working
